### ISSUE TITLE
[HOLD for payment 2023-04-03] Unwanted double quotes in notification after ... for bigger messages

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#14849]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Open the app and login with user A
2. Open the app in android app and login with user B
3. Send long message from user A to user B

## Expected Result:
Larger notifcations should end with ... in the end

## Actual Result:
Larger notification ends with ..." in the end (App is adding unwanted double quotes)

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native

**Version Number:** 1.2.65-0
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 

https://user-images.githubusercontent.com/43996225/216880746-1c0aa913-58e3-4fa1-a43b-8238c1eb6877.mp4

<img src="https://user-images.githubusercontent.com/43996225/216880804-3ac3ef54-b4aa-4715-9107-c8d484350fd3.jpg" width="40%" height="40%">

**Expensify/Expensify Issue URL:**
**Issue reported by:** @dhanashree-sawant
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1675500083078039

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01441300c2ff47e240</li>
        <li>Upwork Job ID: 1624093978140704768</li>
        <li>Last Price Increase: 2023-02-17</li>
    </ul>
</details>

**Comment by zanyrenney** [comment]
Need to try and reproduce! Haven't been able to so far

**Comment by MelvinBot** [comment]
Job added to Upwork: https://www.upwork.com/jobs/~01441300c2ff47e240

**Comment by MelvinBot** [comment]
Current assignee @zanyrenney is eligible for the External assigner, not assigning anyone new.

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor-plus team member for initial proposal review - @parasharrajat (`External`)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @johnmlee101 (`External`), see https://stackoverflow.com/c/expensify/questions/7972 for more details.

**Comment by alexxxwork** [comment]
I've posted a Proposal for similar problem.
To say short I suppose this logic is on the airship service side
https://github.com/Expensify/App/issues/14715#issuecomment-1414433837

**Comment by alexxxwork** [comment]
> @alexxxwork That can not be counted as a proposal as it does not propose any solutions with the analysis.

Yeah, of course.  But I think the most right thing to do is to address the problem to airship service. It would be a hard task to write rules for notification messages truncation on app side without thorough knowledge of how this work inside the airship service.

**Comment by alexxxwork** [comment]
> Ok, @alexxxwork. IMO, Maybe this can easily be resolved from the app side with the proper configuration of the notification builder. I don't think we need to know anything about airship service. the Logic should already be there in the source.

@parasharrajat ok, we could go this way - this way we should count symbols on our side and truncate string sent to notification service. I'll write down a proposal by template.

**Comment by zanyrenney** [comment]
thanks for communicating @parasharrajat - still waiting for proposals here!

**Comment by alexxxwork** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
We are trying to resolve an issue in long string coming to notifications on android native. Strings more than 115 ascii symbols get truncated with `...''` in the end.

### What is the root cause of that problem?
The root cause of the problem is limits of the airship service for android plaform. [doc](https://support.airship.com/hc/en-us/articles/213491643-What-are-the-Maximum-Characters-for-Push-Notifications-) 


### What changes do you think we should make in order to solve the problem?

We should truncate string on our side here
https://github.com/Expensify/App/blob/5a5b788afd91b765bab9d2dfd5de7d3468f0a216/android/app/src/main/java/com/expensify/chat/customairshipextender/CustomNotificationProvider.java#L84


### What alternative solutions did you explore? (Optional)
We could address this issue upstream to airship

**Comment by alexxxwork** [comment]
> Now sure, I understand the proposal.
> 
> * How can you trim the message?
> * How will it solve the problem? What is the run-through?
> 
> Don't we have any API on the builder or something to customize the notification format? I like your optional solution. Do you want to raise it with them?

@parasharrajat We should count symbols in notification string taking in account that airship uses different encodings (we should analyze every symbol) and truncate it on our side in desired format so it doesn't truncated on airship side.
I believe it's better to raise Airship attention on the problem because the above solution is only a workaround.

**Comment by alexxxwork** [comment]
> Yes, I don't think this is the right way to trim the message. It might cause other issues and also what you are proposing has maintains cost to it.

@parasharrajat So how should I address the issue to airship support? I suppose it's a commercial contract?

**Comment by alexxxwork** [comment]
> I think we can start with creating an issue on their repo.

@parasharrajat Airship react native readme (https://github.com/urbanairship/react-native-airship) says we should address issues to https://support.airship.com/. Should I make a service request?

**Comment by MelvinBot** [comment]
@johnmlee101 @parasharrajat @zanyrenney this issue was created 2 weeks ago. Are we close to approving a proposal? If not, what's blocking us from getting this issue assigned? Don't hesitate to create a thread in #expensify-open-source to align faster in real time. Thanks!

**Comment by johnmlee101** [comment]
Waiting for the ticket confirmation about the bug!

**Comment by alexxxwork** [comment]
@parasharrajat @johnmlee101 
I've made some log investigation and found out that every message from airship server has a truncated field (with extra `''`) com.urbanairship.push.ALERT and there's full message in payload.
In this scenario I sent messages from web and received them on android hardware device, watching logs in android studio.
```
2023-02-22 00:39:59.578 17245-18660 New Expensify - UALib   com.expensify.chat                   V  PreferenceDataStore - Saving preference: com.urbanairship.push.LAST_CANONICAL_IDS value: ["7749ee50-b213-11ed-8c09-0242d4a8d64f","54b98200-b21e-11ed-86f2-024295e9d6d7","77435da1-b21e-11ed-8cac-0242b28558ca","9ea15690-b21e-11ed-ad50-0242f634f6e4","b3d6df80-b21e-11ed-8cac-0242b28558ca","34f22c50-b21f-11ed-9bab-0242324b2584","ebc6a2e1-b21e-11ed-bf50-0242a2b5b483","0269c590-b21f-11ed-8874-0242b38fb08d","0f0fd6e1-b21f-11ed-a6b1-024234499bf7","85a50c31-b21f-11ed-b53e-02428a9146a9"]
2023-02-22 00:39:59.583 17245-18659 New Expensify - UALib   com.expensify.chat                   E  Notification channel 4148003058225040 does not exist. Falling back to com.urbanairship.default
2023-02-22 00:39:59.584 17245-18659 New Expensify - UALib   com.expensify.chat                   V  NotificationIdGenerator - Incrementing notification ID count
2023-02-22 00:39:59.585 17245-18659 New Expensify - UALib   com.expensify.chat                   V  NotificationIdGenerator - Notification ID: 1010
2023-02-22 00:39:59.586 17245-18662 New Expensify - UALib   com.expensify.chat                   V  PreferenceDataStore - Saving preference: com.urbanairship.push.LAST_RECEIVED_METADATA value: eyJ2ZXJzaW9uX2lkIjoxLCJ0aW1lIjoxNjc3MDA4Mzk3NTEzLCJwdXNoX2lkIjoiODVhNTBjMzEtYjIxZi0xMWVkLWI1M2UtMDI0MjhhOTE0NmE5IiwiY2FtcGFpZ25zIjp7ImNhdGVnb3JpZXMiOltdfX0=
2023-02-22 00:39:59.589 17245-18659 New Expensify - UALib   com.expensify.chat                   D  IncomingPushRunnable - Received result status 0 for push message: {fromExpensify=true, com.urbanairship.metadata=eyJ2ZXJzaW9uX2lkIjoxLCJ0aW1lIjoxNjc3MDA4Mzk3NTEzLCJwdXNoX2lkIjoiODVhNTBjMzEtYjIxZi0xMWVkLWI1M2UtMDI0MjhhOTE0NmE5IiwiY2FtcGFpZ25zIjp7ImNhdGVnb3JpZXMiOltdfX0=, com.urbanairship.push.APID=0bd51415-27ec-48c4-a8cc-7767b80281d7, payload={"title":"spam@vcc.vc","type":"reportComment","reportID":4148003058225040,"shouldScrollToLastUnread":true,"roomName":null,"onyxData":[{"onyxMethod":"merge","key":"report_4148003058225040","value":{"reportID":"4148003058225040","maxSequenceNumber":42,"lastActionCreated":"2023-02-21 19:39:56.640","lastVisibleActionCreated":"2023-02-21 19:39:56.640","lastReadTime":"2023-02-21 19:35:11.928","lastMessageText":"01234567890123456789012345678901234567890123456789012345678901234567890123456789","lastActorEmail":"spam@vcc.vc"}},{"onyxMethod":"merge","key":"reportActions_4148003058225040","value":{"1819390685033551052":{"person":[{"type":"TEXT","style":"strong","text":"spam@vcc.vc"}],"actorEmail":"spam@vcc.vc","actorAccountID":13031568,"message":[{"type":"COMMENT","html":"0123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789","text":"0123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789","isEdited":false}],"originalMessage":{"html":"0123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789"},"avatar":"https:\/\/d2k5nsl2zxldvw.cloudfront.net\/images\/avatars\/avatar_4.png","created":"2023-02-21 19:39:56.640","timestamp":1677008396,"reportActionTimestamp":1677008396640,"automatic":false,"sequenceNumber":42,"actionName":"ADDCOMMENT","shouldShow":true,"clientID":"","reportActionID":"1819390685033551052","isAttachment":false}},"shouldNotify":true}]}, com.urbanairship.push.CANONICAL_PUSH_ID=85a50c31-b21f-11ed-b53e-02428a9146a9, com.urbanairship.push.PUSH_ID=85b22b90-b21f-11ed-818f-0242b2c0f5a9, com.urbanairship.notification_channel=4148003058225040, com.urbanairship.push.ALERT=0123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234...", com.urbanairship.title=spam@vcc.vc}
2023-02-22 00:39:59.593 17245-18659 New Expensify - UALib   com.expensify.chat                   I  Posting notification: Notification(channel=CHANNEL_MESSAGES shortcut=null contentView=null vibrate=null sound=null defaults=0x0 flags=0x10 color=0xff2eaae2 category=msg vis=PUBLIC semFlags=0x0 semPriority=0 semMissedCount=0) id: 1010 tag: null
```
So messages are truncated server-side in a wrong manner, but there's full message in payload

**Comment by alexxxwork** [comment]
> Waiting for the ticket confirmation about the bug!

@johnmlee101 @parasharrajat 
https://support.airship.com/hc/en-us/requests/89947

**Comment by MelvinBot** [comment]
📣 @alexxxwork! 📣

Hey, it seems we don’t have your contributor details yet! You'll only have to do this once, and this is how we'll hire you on Upwork.
Please follow these steps:

1. Get the email address used to login to your Expensify account. If you don't already have an Expensify account, create one [here](https://new.expensify.com/). If you have multiple accounts (e.g. one for testing), please use your main account email.
2. Get the link to your Upwork profile. It's necessary because we only pay via Upwork. You can access it by logging in, and then clicking on your name. It'll look like [this](https://www.upwork.com/freelancers/~01ca8ad8ce36b9eceb). If you don't already have an account, sign up for one [here](https://www.upwork.com/nx/signup/?dest=home).
3. Copy the format below and paste it in a comment on this issue. Replace the placeholder text with your actual details.

![Screen Shot 2022-11-16 at 4 42 54 PM](https://user-images.githubusercontent.com/30609178/202326034-1a292e64-48fd-4fef-9f4f-7feaaeedfd9f.png)

Format:
```
Contributor details
Your Expensify account email: <REPLACE EMAIL HERE>
Upwork Profile Link: <REPLACE LINK HERE>
```

**Comment by alexxxwork** [comment]
Contributor details
Your Expensify account email: a@vcc.vc
Upwork Profile Link: https://www.upwork.com/freelancers/~01558071651a8abfb5

**Comment by MelvinBot** [comment]
✅ Contributor details stored successfully. Thank you for contributing to Expensify!

**Comment by alexxxwork** [comment]
> @alexxxwork Please share updates regularly on the progress of ticket.

@parasharrajat I have provided requested additional info and they are investigating the issue.

**Comment by fabioh8010** [comment]
Hi, I'm Fábio from Callstack and I would like to take a look at this issue.

**Comment by alexxxwork** [comment]
@parasharrajat not much. I've provided some additional info on the ticket like AndroidManifest.xml, Airshipconfig.plist and so on. They asked to test push message truncation on their test app. Working on it now.

**Comment by zanyrenney** [comment]
Thanks for jumping in here @fabioh8010 !

**Comment by MelvinBot** [comment]
Current assignee @parasharrajat is eligible for the Internal assigner, not assigning anyone new.

**Comment by zanyrenney** [comment]
hey @parasharrajat - that's done now!

**Comment by zanyrenney** [comment]
What is the latest here @johnmlee101 ?

**Comment by Julesssss** [comment]
I'm taking over as this is an internal server issue that I'm going to resolve as part of [this wider task](https://github.com/Expensify/App/issues/14715?).

**Comment by zanyrenney** [comment]
Sounds good @Julesssss - thank you for the update!

**Comment by alexxxwork** [comment]
> We also agreed to compensate you for the same [#14715 (comment)](https://github.com/Expensify/App/issues/14715#issuecomment-1446606033) on another issue.

@parasharrajat Could you please explain how this should work? Should I just wait for these issues to be resolved?

**Comment by zanyrenney** [comment]
What's the latest here @Julesssss ?

**Comment by Julesssss** [comment]
Basically, **this** issue is already fixed, but I can't merge it yet. 

![Screenshot_20230313-154312](https://user-images.githubusercontent.com/10736861/224754495-29371f64-12a5-48ed-82f9-8f36629bb425.png)

It modifies a lot of existing logic that handles notifications for OldDot, so I need to be very careful about the fix and roll out. I am migrating to a different method of building notifications, so we'll need to deploy the new method, but continue to be backward compatible for older users before making the final backend change.

Maybe we should hold this one in the meantime to avoid needing daily updates -- because they won't really be related to THIS issue.

**Comment by zanyrenney** [comment]
Thanks for that!, So we're in a holding pattern here.

**Comment by Julesssss** [comment]
[PR in review](https://github.com/Expensify/App/pull/15857)

**Comment by MelvinBot** [comment]
`Reviewing` label has been removed, please complete the "BugZero Checklist".

**Comment by MelvinBot** [comment]
The solution for this issue has been :rocket: deployed to production :rocket: in version 1.2.89-0 and is now subject to a 7-day regression period :calendar:. Here is the list of pull requests that resolve this issue:

- https://github.com/Expensify/App/pull/15857


If no regressions arise, payment will be issued on 2023-04-03. :confetti_ball:

After the hold period is over and BZ checklist items are completed, please complete any of the applicable payments for this issue, and check them off once done.
- [ ] External issue reporter
- [ ] Contributor that fixed the issue
- [ ] Contributor+ that helped on the issue and/or PR

As a reminder, here are the bonuses/penalties that should be applied for any External issue:
- Merged PR within 3 business days of assignment - 50% **bonus**
- Merged PR more than 9 business days after assignment - 50% **penalty**

**Comment by MelvinBot** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [@parasharrajat / @Julesssss] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@parasharrajat / @Julesssss] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@parasharrajat / @Julesssss] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [@zanyrenney] Determine if we should create a regression test for this bug.
- [ ] [@parasharrajat] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
- [ ] [@zanyrenney] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by Julesssss** [comment]
Hey @zanyrenney, I checked off the boxes [here](https://github.com/Expensify/App/issues/14849#issuecomment-1484878597) on your behalf as this wasn't a regression. 

We are now ready for payment 🎉

**Comment by zanyrenney** [comment]
Hmm weird, this was closed on March 19th.

**Comment by zanyrenney** [comment]
hey @Julesssss I will DM you as there are a few odd things about this one

**Comment by zanyrenney** [comment]
I've chatted to @Julesssss about this one. Seeing as this issue is a dupe of an internal issue that has been fixed as a result of https://github.com/Expensify/App/issues/14849#issuecomment-1461761472 we'll just be paying out the $250 reporting bonus and the $250 for reviews as we appreciate the time taken to do those reviews and the report but was not due the full payment due to being resolved elsewhere.

**Comment by zanyrenney** [comment]
@parasharrajat @dhanashree-sawant invited you both to the job - https://www.upwork.com/jobs/~013b63b7c27ad5bd41

**Comment by dhanashree-sawant** [comment]
Thanks @zanyrenney , I have accepted the invite.

**Comment by Julesssss** [comment]
Hi @alexxxwork. As we didn't assign you to **this** issue, I don't think so. However, you seem to be eligible for payment on [this linked issue](https://github.com/Expensify/App/issues/14715#issuecomment-1446606033). Which is yet to be completely resolved.

**Comment by dhanashree-sawant** [comment]
Hi @zanyrenney, thanks for the offer on upwork but when I tried to accept it now, it says that job is no longer available, can you have a look once you are available?

**Comment by Puneet-here** [comment]
Hi @zanyrenney, this bug was originally reported by me [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1666038366204659)

**Comment by MelvinBot** [comment]
📣 @Puneet-here! 📣

Hey, it seems we don’t have your contributor details yet! You'll only have to do this once, and this is how we'll hire you on Upwork.
Please follow these steps:

1. Get the email address used to login to your Expensify account. If you don't already have an Expensify account, create one [here](https://new.expensify.com/). If you have multiple accounts (e.g. one for testing), please use your main account email.
2. Get the link to your Upwork profile. It's necessary because we only pay via Upwork. You can access it by logging in, and then clicking on your name. It'll look like [this](https://www.upwork.com/freelancers/~01ca8ad8ce36b9eceb). If you don't already have an account, sign up for one [here](https://www.upwork.com/nx/signup/?dest=home).
3. Copy the format below and paste it in a comment on this issue. Replace the placeholder text with your actual details.

![Screen Shot 2022-11-16 at 4 42 54 PM](https://user-images.githubusercontent.com/30609178/202326034-1a292e64-48fd-4fef-9f4f-7feaaeedfd9f.png)

Format:
```
Contributor details
Your Expensify account email: <REPLACE EMAIL HERE>
Upwork Profile Link: <REPLACE LINK HERE>
```

**Comment by MelvinBot** [comment]
@Julesssss, @parasharrajat, @zanyrenney Whoops! This issue is 2 days overdue. Let's get this updated quick!

**Comment by MelvinBot** [comment]
@Julesssss, @parasharrajat, @zanyrenney Whoops! This issue is 2 days overdue. Let's get this updated quick!

**Comment by zanyrenney** [comment]
Hmm, sorry I am confused now. @[dhanashree-sawant](https://github.com/dhanashree-sawant)
dhanashree-sawant i thought you were the reporter. If [Puneet-here](https://github.com/Puneet-here) is the reporter, then you wouldn't get the payment. Can you explain what you're due payment for in this case @dhanashree-sawant ?

**Comment by dhanashree-sawant** [comment]
Hi @zanyrenney, I agree, Puneet had raised the concern but it was 5 months back from this report and we never went ahead with it until I saw the issue again and reported it (I was not aware of the report as it was in expensify-open-source platform until Puneet pointed it out). Its on you on how to handle the payments.

**Comment by Julesssss** [comment]
Ah right. Seems like a fair mistake.

**Comment by Julesssss** [comment]
This is a tough call. Given that no further action was taken on the original issue I think the newer bug report should receive the bounty, personally. It looks like it was simply missed originally :/

**Comment by zanyrenney** [comment]
Thanks @Julesssss I agree.

**Comment by dhanashree-sawant** [comment]
Hi @zanyrenney,

Payment for the reporting bonus is left.

**Comment by kbecciv** [comment]
Re0peing the issue
Slack conversation: https://expensify.slack.com/archives/C049HHMV9SM/p1694364092518829

**Comment by Julesssss** [comment]
This is

> Reopeing the issue Slack conversation: https://expensify.slack.com/archives/C049HHMV9SM/p1694364092518829

Can you please open a new issue and assign me? This one is 6 months old and while similar, it will require a different solution

**Comment by kbecciv** [comment]
Sure, will do that @Julesssss

**Comment by kbecciv** [comment]
Link new issue  https://github.com/Expensify/App/issues/27663
